src/variables.ts

```diff
@@ -17,15 +17,19 @@ export class SitewiseVariableSupport extends CustomVariableSupport<DataSource, S
   editor = QueryEditor;
 
   query(request: DataQueryRequest<SitewiseQuery>): Observable<DataQueryResponse> {
-    assign(request.targets, [{ ...request.targets[0], refId: 'A' }]);
-    const response = this.datasource.query(request);
-    switch (request.targets[0].queryType) {
-      case QueryType.ListAssetModels:
-      case QueryType.ListAssets:
-      case QueryType.ListAssociatedAssets:
-        return this.parseOptions(response);
-      default:
-        return response;
+    if (this.isValidQuery(request.targets[0])) {
```

Comment:
Nice! This should prevent unnecessary invocation of `query(...)`. 👍 
Curious if there's another place/interface to achieve this in Grafana?